src/main/java/software/amazon/event/kafkaconnector/logging/GitCommitAwareLoggerFactory.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package software.amazon.event.kafkaconnector.logging;
+
+import java.io.IOException;
+import java.util.Objects;
+import java.util.Properties;
+import org.apache.kafka.common.utils.LogContext;
+import org.slf4j.Logger;
+
+public class GitCommitAwareLoggerFactory {
+
+  private static final LogContext context;
+
+  static {
+    var properties = new Properties();
+    try {
+      properties.load(
+          Objects.requireNonNull(GitCommitAwareLoggerFactory.class.getResource("/git.properties"))
+              .openStream());
+    } catch (NullPointerException | IOException ignore) {
+    }
+    var gitCommitIdDescribe = properties.getOrDefault("git.commit.id.describe", "unknown");
```

Comment:
@maschnetwork we decided to use `unknown` in case the git commit cannot be retrieved. WDYT? `unknown-git-rev` more clear?

Reply:
cc/ @baldawar

Reply:
If a valid git commit is found, logging output will be for example: https://github.com/awslabs/eventbridge-kafka-connector/actions/runs/7512809727/job/20472261500?pr=205#step:4:1575 (E2E)

Reply:
`unknown-git-rev` is more clear.